subsys/net/ip/net_context.c

```diff
@@ -129,6 +129,13 @@ int net_context_get(sa_family_t family,
 		return -EPFNOSUPPORT;
 	}
 #endif
+#if !defined(CONFIG_SOCKET_CAN)
```

Comment:
Hi, 
Thanks for the comments. 
To add PF_CAN protocol, it needs to re-use existing network layer of Zephyr. According to CAN CIA ORG it is mentioned that "CAN network device drivers
implement the same standardized networking driver model as Ethernet drivers".
Reference:  https://www.can-cia.org/fileadmin/resources/documents/proceedings/2012_kleine-budde.pdf,

Reply:
I don't think a document for Linux applies for Zephyr.

Reply:
@pfalcon In order to work with socket, CAN needs to integrate with net_if, net_context, having an l2 etc... thus why it has stuff in net. Also, in a "short future", socket won't be a library anymore but part of the net core (thus why at some point, net/ip is going to be more than ip stuff only).

Reply:
> In order to work with socket, CAN needs to integrate with net_if, net_context, having an l2 etc...

Not really - see e.g. how TLS sockets are implemented, they didn't patch anything in net/ip/. And @rlubos now does great work of refactoring socket dispatching to allow even easier integration of different socket types (as part of https://github.com/zephyrproject-rtos/zephyr/pull/11438).

> Also, in a "short future", socket won't be a library anymore but part of the net core (thus why at some point, net/ip is going to be more than ip stuff only).

Not the way I see it. As I see it, net_context is already a *legacy implementation detail* of TCP/UDP networking. In some (not so near) future, when net_context is deprecated from being a public interface, it can change drastically, to support more efficient TCP/UDP networking for sockets. But sockets will remain a separate multiplexing layer on top of it, exactly because of the need to support disparate protocol implementations thru them. These protocol implementations should remain separate however, and not lumped one into another.